### DISCUSSION TITLE
Using a time interval to reassess a condition

### DISCUSSION THREAD

**Discussion by Johntron1219** [bonsai-rx/bonsai#995] [Q&A]
Hi Everyone, 

I was wondering if anyone has experience or ideas for conditional circuits using Bonsai. Here's what I am trying to do: 

An event is recorded in a CSV if 2 conditions are met: 
1. Color A is present in a region of interest and 
2. Color B is present in a different region of interest

Now, I've gotten the function to work, except in one semi-common instance. What keeps happening is that color A will be present in the region for a split-second, and then disappear, while Color B is present in its own region. What I need to do is basically ask: is Color A present? Ok, then is color B present? (wait half a second) Is color A present still, or was the overlap a fluke? 

Does anyone have any ideas for a timed circuit I could build? Every time I try a timespan to reassess it won't take the arguments. 

Thanks, and I hope this makes sense :)!
John

**Comment by bruno-f-cruz** [comment]
Hey,

There are a few ways to do this depending on the exact logic you are looking for. If i understood correctly you might just need to use a `Throttle` operator. This node works a bit like a "low-pass" filter for events by only propagating values that are NOT followed by a new value within X amount of time.  As a result, if a value is only transiently `True` (i.e. `True` quickly followed by `False`) the throttle will effectively skip this `True` value.

![image](https://user-images.githubusercontent.com/7049351/183352410-9eb67705-6168-4fc3-adc0-4c531f5f6588.png)


hope it helps

**Reply by Johntron1219** [reply]
Thank you Bruno! 

I ended up revising my file to run in parallel 20 frames ahead, but I will keep this in mind as a future option!
